test/e2e/major_upgrade_v4.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/celestiaorg/knuu/pkg/knuu"
+
+	"github.com/celestiaorg/celestia-app/v4/app"
+	"github.com/celestiaorg/celestia-app/v4/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/v4/test/e2e/testnet"
+)
+
+func MajorUpgradeToV4(logger *log.Logger) error {
```

Comment:
the test itself is almost identical to the previous upgrade tests that were removed. just specifying an upgrade version of 4, starting on consesnsus v3 and using the multiplexer.